test/integration/integration_helpers.go

```diff
@@ -263,3 +264,17 @@ func execTemplateOnFile(path, tmplStr string, o interface{}) error {
 	}
 	return tmpl.Execute(w, o)
 }
+
+func mkTempDirWithCleanup(t *testing.T, prefix string) (dir string, f func()) {
+	var err error
+	if dir, err = ioutil.TempDir("", prefix); err != nil {
+		t.Fatalf("Failed to create tmp dir: %v", err)
+	}
+	f = func() {
+		if err := os.RemoveAll(dir); err != nil {
+			// Not a test failure since files in /tmp will eventually get deleted
+			t.Logf("Failed to remove tmp dir %s: %v", dir, err)
+		}
+	}
+	return
```

Comment:
Do we do this often? empty return and expect it returns the 2 things? I prefer `return dir, f` since it is explicit and I don't have to think about it.

Reply:
Not often and not for user-facing/long functions. This is slightly more readable to me.